lucee-java/lucee-core/src/lucee/commons/io/res/type/file/FileResource.java

```diff
@@ -74,7 +75,11 @@ public void copyFrom(Resource res,boolean append) throws IOException {
 
 	@Override
 	public void copyTo(Resource res,boolean append) throws IOException {
-		IOUtil.copy(this, res.getOutputStream(append),true);
+		if (res instanceof S3Resource && !append) {

```

Comment:
This optimization prevents two additional file copies (one in `res.getOutputStream` and another within the `S3Fascade` with our new code)